site/content/blog/2020-09-07-improving-performance-through-load-testing.md

```diff
@@ -0,0 +1,83 @@
+
+![actions-map](/blog/2020-09-07-improving-performance-through-load-testing/actions-map.png)
+
+In such an example, the chance of a user switching a channel will be double the chance of creating a new post and four times that of replying to a thread.
+
+### Getting some answers
+
+Now that we have a simulation in place, we come back to the original question of this blog post: how can we make our load testing software tell us how many concurrently active users our target application can support?
+
+Instead of relying on [magic formulas](https://en.wikipedia.org/wiki/Magic_number_(programming)) which are hard to explain (and sometime defend) or on endless and error-prone manual testing, we thought of picking an idea out of [control theory](https://en.wikipedia.org/wiki/Control_theory) that could give us both automation and precision at the same time: it's called a **feedback loop**.
+
+![feedback-loop](/blog/2020-09-07-improving-performance-through-load-testing/feedback-loop.png)
+
+The idea is quite simple: have the load testing engine react to what's happening on the target system by dynamically changing the amount of load applied in response to some event. This is achieved through the monitoring of those metrics that can best signal service degradation. Some that we make use of are:
+
+- Request durations (average and percentiles)
+- Request errors rate
+- Client timeouts rate
+
+![graph](/blog/2020-09-07-improving-performance-through-load-testing/graph.png)
+
+The load testing software constantly increases the number of active users up until it detects signs of performance degradation. This happens when the metrics being monitored reach their configured thresholds (e.g. 100ms for average request duration).
+In response to such an event, the load testing engine starts to lower the number of users. This process continues for a while until an **equilibrium point** is reached. This value, over which performance starts to degrade, gives us a fairly good estimate on the user capacity our target application can comfortably support.
+
+## Wrapping up
+
+The success of a software system greatly depends on its performance. In today's cloud-oriented world this is increasily more true. A small outage can have a serious impact on your customers and eventually affect your company's revenue, especially when a [service level agreement](https://en.wikipedia.org/wiki/Service_level_agreement) is in place.
+
+Load testing is a precious tool that can help your team to improve both the overall quality and stability of your system by preventing performance issues and application bugs while improving scalability and minimizing the risk of down times.
+
```

Comment:
Do we have any future plans or ideas? If so we can add here to initiate a discussion.

Reply:
I'll add a teaser at the end that hopefully will keep more technical folks interested.

Reply:
🕺